sdk/python/feast/dqm/utils.py

```diff
@@ -0,0 +1,73 @@
+from typing import TYPE_CHECKING, List, Optional
+
+import pandas as pd
+import pyarrow
+
+from feast.dqm.errors import ValidationFailed
+from feast.dqm.profilers.profiler import Profile
+from feast.infra.offline_stores.offline_store import RetrievalJob, RetrievalMetadata
+from feast.saved_dataset import SavedDatasetStorage, ValidationReference
+
+if TYPE_CHECKING:
+    from feast.feature_store import FeatureStore
+    from feast.on_demand_feature_view import OnDemandFeatureView
+
+
+class RetrievalJobWithValidation(RetrievalJob):
```

Comment:
I am really not a big fan of maintaining 2 objects "RetrievalJob" that are that close ... 
sounds like and optional parameter `validation_reference` in the current RetrievalJob would be better IMO

Reply:
Also we should be able to use DQM and ODFV together. So maintaining that logic in the same object looks more long term to me.

My expectation would be if DQM and ODFV are used together, it would be processed as follow
1. Run `odfv.get_transformed_features_df()` for each ODFV transformation
2. Validate the dataset against its reference
3. Return the dataset

Steps 1. and 2. are optional to me and can be used alone and together.

WDYT ?

Reply:
agreed with Matt on having one implementation

Reply:
Updated as suggested.